### PR TITLE
Adding depends-on block for repository ownership.

### PR DESCRIPTION
**Pull request by julialawrence**
When destroying repos, we are removing ourselves as owners before we have a chance to destroy the repos, thus blocking ourselves. This adds a depends_on block which should theoretically solve this issue.

### CODE REVIEW

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]
Should we make `archive_on_delete` match the actual flag `archive_on_destroy`?

**Comment by julialawrence** [comment]
> Should we make `archive_on_delete` match the actual flag `archive_on_destroy`?

Yes.

File: terraform/github/main.tf [hunk 1]
```diff
@@ -165,4 +166,5 @@ module "data_platform_labs_team" {
   maintainers = local.data_platform_labs_maintainers
   members     = local.all_members_data_platform_labs
   ci          = local.ci_users
+  depends_on  = [module.data-platform]
```
Comment (tom-webber) [inline terraform/github/main.tf]: Why is this dependency with the `data_platform_labs_team` module? Is it arbitrary? Could it sit within any team module?

**Review by jacobwoffenden — DISMISSED** [review]
~LGTM~ 👀

File: terraform/github/main.tf (hunk 1 above)
Comment (julialawrence) [reply terraform/github/main.tf]: Ugh....

**Review by jacobwoffenden — APPROVED** [review]
LGTM

**Pull request merged**